src/main/java/org/jenkinsci/plugins/githubreleaseperformer/GitHubReleasePerformer.java

```diff
@@ -0,0 +1,225 @@
+package org.jenkinsci.plugins.githubreleaseperformer;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.tokenmacro.TokenMacro;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHReleaseBuilder;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+
+public class GitHubReleasePerformer extends Builder {
+
+	private final String apiUrl;
+	private final String tag;
+	private final String branch;
+	private final String releaseNotesFile;
+	private final String user;
+	private final String password;
+	private final String owner;
+	private final String repository;
+
+	private GitHub github;
+	private GHRepository ghRepository;
+
+	private static final String RELEASE_NOTES_TEMPLATE =
```

Comment:
One could always edit release note after release. :shit: Shit happens. However whatever we have in release notes file(Breaking changes, Migration stuff, Fixes,...) by the time of release should be the content that we want on the new release, because the content of release notes just reflects the code that we release. Also release from tag/branch is also supported.